src/Aspire.Hosting/ApplicationModel/IValueProvider.cs

```diff
@@ -3,6 +3,27 @@
 
 namespace Aspire.Hosting.ApplicationModel;
 
+/// <summary>
+/// Provides context for value resolution.
+/// </summary>
+public class ValueProviderContext
+{
+    /// <summary>
+    /// Additional services that can be used during value resolution.
+    /// </summary>
+    public IServiceProvider? Services { get; init; }
+
+    /// <summary>
+    /// The resource that is requesting the value.
+    /// </summary>
+    public IResource? Caller { get; init; }
+
+    /// <summary>
+    /// The identifier of the network that serves as the context for value resolution.
+    /// </summary>
+    public NetworkIdentifier? Network { get; init; }
```

Comment:
I assume we expect this value to not be set in certain scenarios. For example, I'd expect this to be null during deployment mode. If so, how do we think about modeling the ValueProviderContext under different execution contexts?